ext/nnstreamer/tensor_converter/tensor_converter_flatbuf.cc

```diff
@@ -0,0 +1,169 @@
+
+namespace nnstreamer {
+namespace flatbuf {
+
+void init_fbc (void) __attribute__ ((constructor));
+void fini_fbc (void) __attribute__ ((destructor));
+
+/** @brief tensor converter plugin's NNStreamerExternalConverter callback */
+static GstCaps *
+fbc_query_caps (const GstTensorsConfig * config)
+{
+  return gst_caps_from_string (GST_FLATBUF_TENSOR_CAP_DEFAULT);
+}
+
+/** @brief tensor converter plugin's NNStreamerExternalConverter callback */
+static gboolean
+fbc_get_out_config (const GstCaps * in_cap, GstTensorsConfig * config)
+{
+  GstStructure *structure;
+  g_return_val_if_fail (config != NULL, FALSE);
+  gst_tensors_config_init (config);
+  g_return_val_if_fail (in_cap != NULL, FALSE);
+
+  structure = gst_caps_get_structure (in_cap, 0);
+  g_return_val_if_fail (structure != NULL, FALSE);
+
+  /* All tensor info should be updated later in chain function. */
+  config->info.info[0].type = _NNS_UINT8;
+  config->info.num_tensors = 1;
+  if (gst_tensor_parse_dimension ("1:1:1:1",
+          config->info.info[0].dimension) == 0) {
+    ml_loge ("Failed to set initial dimension for subplugin");
+    return FALSE;
+  }
+
+  if (gst_structure_has_field (structure, "framerate")) {
+    gst_structure_get_fraction (structure, "framerate", &config->rate_n,
+        &config->rate_d);
+  } else {
+    /* cannot get the framerate */
+    config->rate_n = 0;
+    config->rate_d = 1;
+  }
+  return TRUE;
+}
+
+/** @brief tensor converter plugin's NNStreamerExternalConverter callback
+ *  @todo : Consider multi frames, return Bufferlist and 
+ *          remove frame size and the number of frames
+ */
+static GstBuffer *
+fbc_convert (GstBuffer * in_buf, gsize * frame_size, guint * frames_in,
+    GstTensorsConfig * config)
+{
+  const Tensors *tensors;
+  const flatbuffers::Vector < flatbuffers::Offset < Tensor >> *tensor;
+  const flatbuffers::Vector < unsigned char >*tensor_data;
+  frame_rate fr;
+  GstBuffer *out_buf;
+  GstMemory *in_mem, *out_mem;
+  GstMapInfo in_info;
+  guint mem_size;
+  gpointer mem_data;
+
+  in_mem = gst_buffer_peek_memory (in_buf, 0);
+  g_assert (gst_memory_map (in_mem, &in_info, GST_MAP_READ));
+
+  tensors = GetTensors (in_info.data);
+  g_assert (tensors);
+
+  config->info.num_tensors = tensors->num_tensor ();
+  config->rate_n = tensors->fr ()->rate_n ();
+  config->rate_d = tensors->fr ()->rate_d ();
+
+  tensor = tensors->tensor ();
+
+  out_buf = gst_buffer_new ();
+
+  for (guint i = 0; i < config->info.num_tensors; i++) {
+    config->info.info[i].name =
+        g_strdup (tensor->Get (i)->name ()->str ().c_str ());
+    config->info.info[i].type = (tensor_type) tensor->Get (i)->type ();
+    tensor_data = tensor->Get (i)->data ();
+
+    for (guint j = 0; j < NNS_TENSOR_RANK_LIMIT; j++) {
+      config->info.info[i].dimension[j] =
+          tensor->Get (i)->dimension ()->Get (j);
+    }
+    *frame_size = mem_size = VectorLength (tensor_data);
+    *frames_in = 1;
```

Comment:
Could you explain the way to set `frame_size` and `frames_in` in the above two lines?

Reply:
The data type of tensor_data is ubyte(uint8) which is defined in nnstreamer.fbs. So memory size is matched with the length of the vector. (Flatbuffer handles data using vector)
And tensor converter supports a single frame only currently, so I just set 1.

Reply:
Thanks for the explanation. I understood `frames_in`.

But, in case of `frame_size`, it will be updated for every iteration, and finally set to the last tensor's vector length. Is it intended behavior?

Reply:
You're right. I updated frame_size. Thank you!